src/fidesops/service/privacy_request/request_runner_service.py

```diff
@@ -225,6 +211,47 @@ def run(
             logging.info(f"Privacy request {privacy_request.id} run completed.")
             session.close()
 
+    @staticmethod
+    def upload_access_results(
+        session: Session,
+        policy: Policy,
+        access_result: Dict[str, List[Row]],
+        dataset_graph: DatasetGraph,
+        privacy_request: PrivacyRequest,
+    ) -> None:
+        """Process the data uploads after the access portion of the privacy request has completed"""
+        if not access_result:
+            logging.info(f"No results returned for access request {privacy_request.id}")
+
+        for rule in policy.get_rules_for_action(action_type=ActionType.access):
+            if not rule.storage_destination:
+                raise common_exceptions.RuleValidationError(
+                    f"No storage destination configured on rule {rule.key}"
+                )
+            target_categories: Set[str] = {
+                target.data_category for target in rule.targets
+            }
+            filtered_results = filter_data_categories(
+                access_result,
+                target_categories,
+                dataset_graph.data_category_field_mapping,
+            )
+            logging.info(
+                f"Starting access request upload for rule {rule.key} for privacy request {privacy_request.id}"
+            )
+            try:
+                upload(
+                    db=session,
+                    request_id=privacy_request.id,
+                    data=filtered_results,
+                    storage_key=rule.storage_destination.key,
+                )
+            except common_exceptions.StorageUploadError as exc:
+                logging.error(
+                    f"Error uploading subject access data for rule {rule.key} on policy {policy.key} and privacy request {privacy_request.id} : {exc}"
+                )
+                privacy_request.status = PrivacyRequestStatus.error
+
```

Comment:
I just pulled this out because PrivacyRequestRunner.run is getting long, and it's a specific step that should not run if we're resuming the privacy request from the erasure portion.  There's some logic that's been added to PrivacyRequestRunner.run to allow it to be used to resume from both the access and erasure portions of the request.